### PR TITLE
Update git-clone task to fix custom CAs for https and bump to 0.7 version

### PR DESCRIPTION
**Pull request by savitaashture**
**Issue:** tried cloning secured repository by creating secret for `crt` and mounted using `ssl-ca-directory` but still faced below issue
```
[clone] {"level":"error","ts":1655989849.1085541,"caller":"git/git.go:55","msg":"Error running git [fetch --recurse-submodules=yes --depth=1 origin --update-head-ok --force master]: exit status 128\nfatal: unable to access 'https://gitlab.cee.redhat.com/tekton/plumbing.git/': SSL certificate problem: unable to get local issuer certificate\n","stacktrace":"github.com/tektoncd/pipeline/pkg/git.run\n\tgithub.com/tektoncd/pipeline/pkg/git/git.go:55\ngithub.com/tektoncd/pipeline/pkg/git.Fetch\n\tgithub.com/tektoncd/pipeline/pkg/git/git.go:150\nmain.main\n\tgithub.com/tektoncd/pipeline/cmd/git-init/main.go:53\nruntime.main\n\truntime/proc.go:225"}
```
# Changes
* Added `GIT_SSL_CAINFO`
* Bumped new version `0.7`

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Follows the [authoring recommendations](https://github.com/tektoncd/catalog/blob/main/recommendations.md)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [ ] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [ ] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [ ] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [ ] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [ ] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [ ] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

**Review by vinamra28 — COMMENTED** [review]
overall PR looks good. @savitaashture can we add some docs around the usage of this task as how one can mount the ca-bundle.crt file while using this task

/lgtm
Thanks for fixing this 👼

**Review by vdemeester — APPROVED** [review]

File: task/git-clone/0.6/git-clone.yaml [hunk 1]
```diff
@@ -180,8 +186,12 @@ spec:
           chmod -R 400 "${PARAM_USER_HOME}"/.ssh/*
         fi
 
+        echo "what is the file location"
+        ls -lrt ${WORKSPACE_SSL_CA_DIRECTORY_PATH}
```
Comment (vinamra28) [inline task/git-clone/0.6/git-clone.yaml:190]: do we need this?

File: task/git-clone/0.5/git-clone.yaml [hunk 2]
```diff
@@ -180,8 +186,12 @@ spec:
           chmod -R 400 "${PARAM_USER_HOME}"/.ssh/*
         fi
 
+        echo "what is the file location"
+        ls -lrt ${WORKSPACE_SSL_CA_DIRECTORY_PATH}
```
Comment (vinamra28) [inline task/git-clone/0.5/git-clone.yaml:190]: same

File: task/git-clone/0.6/git-clone.yaml (hunk 1 above)
Comment (savitaashture) [reply task/git-clone/0.6/git-clone.yaml]: Yes added to check the file list in the current directory

Removed

**Comment by chmouel** [comment]
do we have to update 0.5 and 0.6, isnt that policy to bump version (0.7.x) when there is a new params added?

**Comment by savitaashture** [comment]
@vdemeester @chmouel Bumped to new 0.7 version for git-clone

PTAL 
Thank you

**Review by chmouel — COMMENTED** [review]
this lgtm to me, if you wanted to go the extra mile you could add the sample you had in README in the `samples` directory,


/lgtm

**Comment by savitaashture** [comment]
> this lgtm to me, if you wanted to go the extra mile you could add the sample you had in README in the `samples` directory,
> 
> /lgtm

Addressed here https://github.com/tektoncd/catalog/pull/1007

**Pull request merged**